wcfsetup/install/files/js/WCF.Attachment.js

```diff
@@ -308,6 +315,85 @@ if (COMPILER_TARGET_DEFAULT) {
 			// remove failed uploads
 			this._fileListSelector.children('li.uploadFailed').remove();
 			
+			if (this._options.enableAutoScale) {
+				var self = this;
+				var maxSize = this._buttonSelector.data('maxSize');
+				
+				var promise = new Promise(function (resolve) {
+					require(['ImageResizer'], function (ImageResizer) {
+						var promises = [];
+						
+						// Create an ImageResizer instance
+						var resizer = new ImageResizer()
+							.setMaxWidth(self._options.autoScaleMaxWidth)
+							.setMaxHeight(self._options.autoScaleMaxHeight)
+							.setFileType(self._options.autoScaleFileType)
+							.setQuality(self._options.autoScaleQuality);
+						
+						for (var i = 0; i < files.length; i++) {
+							if (files[i].type.match(/^image\//i)) {
+								(function (i) {
+									promises.push(new Promise(function (resolve) {
+										var file = files[i];
+										
+										if (file.blob) {
+											var name = file.name;
+											file = file.blob;
+											file.name = name;
+										}
+										
+										var reader = new FileReader();
+										var image = new Image();
+										var needsResize = file.size > maxSize;
+										
+										reader.onloadend = function () {
+											image.src = reader.result;
+										};
+										
+										image.onload = function () {
+											needsResize = needsResize
+												    || image.width > self._options.autoScaleMaxWidth
+												    || image.height > self._options.autoScaleMaxHeight;
+											
+											if (!needsResize) {
+												resolve(file);
+											}
+											else {
+												resizer.resize(image)
+													.then(function () {
+														return resizer.getFile(file.name)
+													})
+													.then(resolve)
+													.catch(function (error) {
+														// In case of an error return the original file
+														console.debug('[WCF.Attachment] Failed to resize', error);
+														resolve(file);
+													});
+											}
+											
+										};
+										
+										image.onerror = function (event) {
+											console.debug('[WCF.Attachment] Failed to create image object. Your browser probably does not support the image type ('+file.type+').');
+											resolve(file);
+										};
+										
+										reader.readAsDataURL(file);
+									}));
+								})(i);
+							}
+							else {
+								promises.push(Promise.resolve(files[i]));
+							}
+						}
+						
+						resolve(Promise.all(promises));
+					});
```

Comment:
Instead of rejecting the promise I would resolve with the untouched list of files, thus I don’t have to worry about rejected promises in WCF.Upload.
I think that failure to load the required modules is the uncommon case, which makes trying to upload the untouched files sound like a sane failure mode :smile: